tower-http/src/metrics/traffic/future.rs

```diff
@@ -0,0 +1,95 @@
+use super::{Callbacks, FailedAt, ResponseBody};
+use crate::classify::{ClassifiedResponse, ClassifyResponse};
+use futures_core::ready;
+use http::Response;
+use http_body::Body;
+use pin_project::pin_project;
+use std::{
+    fmt,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Instant,
+};
+
+/// Response future for [`Traffic`].
+///
+/// [`Traffic`]: crate::metrics::Traffic
+#[pin_project]
+pub struct ResponseFuture<F, C, Callbacks, CallbacksData> {
+    #[pin]
+    pub(super) inner: F,
+    pub(super) classifier: Option<C>,
+    pub(super) request_received_at: Instant,
+    pub(super) callbacks: Option<Callbacks>,
+    pub(super) callbacks_data: Option<CallbacksData>,
+}
+
+impl<F, C, ResBody, E, CallbacksT, CallbacksData> Future
+    for ResponseFuture<F, C, CallbacksT, CallbacksData>
+where
+    F: Future<Output = Result<Response<ResBody>, E>>,
+    ResBody: Body,
+    C: ClassifyResponse,
+    CallbacksT: Callbacks<C::FailureClass, Data = CallbacksData>,
+    E: fmt::Display + 'static,
+{
+    type Output =
+        Result<Response<ResponseBody<ResBody, C::ClassifyEos, CallbacksT, CallbacksT::Data>>, E>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.project();
+        let result = ready!(this.inner.poll(cx));
+
+        let classifier = this.classifier.take().unwrap();
```

Comment:
Do you think it might make more sense if we replace `unwrap()` with `expect("polled ResponseFuture after completion")`?

This would be more consistent with existing messages provided by the various futures types in the `futures` crate, etc.